config/pipeline.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+module.exports = {
+    AttributeDefinitions: [ /* required */
+        {
+            AttributeName: 'id', /* required */
+            AttributeType: 'S' /* required */
+        },
+        {
+            AttributeName: 'scmUrl', /* required */
+            AttributeType: 'S' /* required */
+        }
+    ],
+    KeySchema: [ /* required */
+        {
+            AttributeName: 'id', /* required */
+            KeyType: 'HASH' /* required */
+        }
+    ],
+    ProvisionedThroughput: { /* required */
+        ReadCapacityUnits: 1, /* required */
+        WriteCapacityUnits: 1 /* required */
+    },
+    TableName: 'Pipelines', /* required */
+    GlobalSecondaryIndexes: [
+        {
+            IndexName: 'ScmUrls', /* required */
+            KeySchema: [ /* required */
+                {
+                    AttributeName: 'scmUrl',
+                    KeyType: 'HASH'
+                }
+            ],
+            Projection: { /* required */
+                ProjectionType: 'KEYS_ONLY'
```

Comment:
https://aws.amazon.com/dynamodb/faqs/

> Q: Can I retrieve all attributes of a DynamoDB table from a global secondary index?
>
> A query on a GSI can only return attributes that were specified to be included in the GSI at creation time. The attributes included in the GSI are those that are projected by default such as the GSI’s key attribute(s) and table’s primary key attribute(s), and those that the user specified to be projected. For this reason, a GSI query will not return attributes of items that are part of the table, but not included in the GSI. A GSI that specifies all attributes as projected attributes can be used to retrieve any table attributes. See here for documentation on using GSIs for queries.

TL;DR - Only the attributes projected in the index will be available when queried. The initial pass was that we only care for uniqueness of `scmUrls` and necessarily any other data.